CMakeLists.txt

```diff
@@ -24,5 +24,7 @@ add_subdirectory(zeq)
 add_subdirectory(tests)
 add_subdirectory(tools)
 
+set(DOXYGEN_MAINPAGE_MD README.md)
+set(DOXYGEN_EXTRA_INPUT ${PROJECT_SOURCE_DIR}/README.md)
```

Comment:
I see. DOXYGEN_EXTRA_INPUT is needed because README.md is not under doc. If all documentation files were under doc no extra variable would be needed. DOXYGEN_MAINPAGE_MD is not needed if you add a title with {#mainpage} to README.md. The funny thing is that doxygen adds a title automatically to the main page if you omit it, so if you have removed it because you think it'd look nicer it won't work as expected.
I'm realizing that we have dozens of variations on how we do things, and it gets confusing.